library/swift/UserDefaultsStore.swift

```diff
@@ -0,0 +1,23 @@
+import Foundation
+
+/// Simple implementation of a `KeyValueStore` leveraging `UserDefaults` for persistence.
+@objcMembers
+public final class UserDefaultsStore: KeyValueStore {
```

Comment:
Why not remove this type altogether and make `UserDefaults` conform directly to `KeyValueStore`?

```swift
extension UserDefaults: KeyValueStore {
  public func readValue(forKey key: String) -> String? {
    self.string(forKey: key)
  }

  public func saveValue(_ value: String, toKey key: String) {
    self.set(value, forKey: key)
  }

  public func removeKey(_ key: String) {
    self.removeObject(forKey: key)
  }  
}
```

Reply:
That's pretty clever, and I'm not opposed to that approach. :) If you'd like to push a PR that does it, I'd be open to closing this in favor it, though otherwise I'd suggest we proceed with this for now.

Reply:
Sure, I'll put something up shortly.

Reply:
https://github.com/envoyproxy/envoy-mobile/pull/2452